src/wasm2js.h

```diff
@@ -503,6 +512,10 @@ void Wasm2JSBuilder::addBasics(Ref ast, Module* wasm) {
     addHeap(HEAPU32, UINT32ARRAY);
     addHeap(HEAPF32, FLOAT32ARRAY);
     addHeap(HEAPF64, FLOAT64ARRAY);
+    ast->push_back(
+      ValueBuilder::makeBinary(ValueBuilder::makeName("bufferView"),
+                               SET,
+                               ValueBuilder::makeName(HEAPU8)));
```

Comment:
this creation of `bufferView` looks unconditional - whenever there is a memory at all. why is another one needed elsewhere?

Reply:
This is not a creation of anything but an assignment to the outer scoped `bufferView`

But you are right I think it needs to be conditional in the same way that bufferView itself is conditionally declared in the outer scope.

Reply:
Oh, I see, I was misreading it, sorry.